api/src/main/java/org/geysermc/geyser/api/event/lifecycle/SkinApplyEvent.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2019-2024 GeyserMC. http://geysermc.org
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @author GeyserMC
+ * @link https://github.com/GeyserMC/Geyser
+ */
+
+package org.geysermc.geyser.api.event.lifecycle;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.geysermc.event.Event;
+import org.geysermc.geyser.api.skin.Cape;
+import org.geysermc.geyser.api.skin.Skin;
+import org.geysermc.geyser.api.skin.SkinData;
+import org.geysermc.geyser.api.skin.SkinGeometry;
+
+import java.util.UUID;
+
+public abstract class SkinApplyEvent implements Event {
```

Comment:
Shouldn't we call it `SessionSkinApplyEvent`?
We prefix all other events atm

Reply:
I wasn't sure as we don't have a `GeyserConnection` like all of the other `Session` prefixed events, we could get a session handle but would require a bunch more work to get hold of it as its not passed through

Reply:
Would this event be also called for Java players? aka when their skin would be applied?

Reply:
Yes this is called when any skin is applied, I believe it will also include things like the fake players we use for skulls

Reply:
I anticipate if the event doesn't indicate a skin is for a player skull, there will be confusion.

Reply:
We could exclude skulls and make sure it's clear in the javadoc that it's for players only

Reply:
Confirmed the event doesn't get called when a player skull is spawned.